aiohttp_jinja2/helpers.py

```diff
@@ -2,27 +2,39 @@
 useful context functions, see
 http://jinja.pocoo.org/docs/dev/api/#jinja2.contextfunction
 """
-from typing import Any, Dict, cast
+import sys
+from typing import Any, Dict, Union
 
 import jinja2
 from aiohttp import web
 from yarl import URL
 
+if sys.version_info >= (3, 8):
+    from typing import TypedDict
+
+    class _Context(TypedDict, total=False):
+        app: web.Application
+
+
+else:
+    _Context = Dict[str, Any]
+
 
 @jinja2.contextfunction
-def url_for(context: Dict[str, Any], __route_name: str, **parts: Any) -> URL:
+def url_for(context: _Context, __route_name: str, **parts: Union[str, int]) -> URL:
     """Filter for generating urls.
 
     Usage: {{ url('the-view-name') }} might become "/path/to/view" or
     {{ url('item-details', id=123, query={'active': 'true'}) }}
     might become "/items/1?active=true".
     """
-    app = cast(web.Application, context["app"])
+    app = context["app"]
 
     query = None
     if "query_" in parts:
-        query = parts.pop("query_")
+        query = str(parts.pop("query_"))
```

Comment:
OK, I missed that this argument is meant to be different from the rest of `parts`. It seems like it's supposed to be a dedicated keyword argument, therefore I have made it a proper keyword argument. Seems to work just fine, and avoids the confusion.